api/base/exceptions.py

```diff
@@ -16,21 +16,20 @@ def json_api_exception_handler(exc, context):
 
     if response:
         message = response.data
+
         if isinstance(message, dict):
-            for key, value in message.iteritems():
-                if key in top_level_error_keys:
-                    errors.append({key: value})
+            for error_key, error_description in message.iteritems():
+                if error_key in top_level_error_keys:
+                    errors.append({error_key: error_description})
                 else:
-                    if isinstance(value, list):
-                        for reason in value:
-                            errors.append({'detail': reason, 'meta': {'field': key}})
-                    else:
-                        errors.append({'detail': value, 'meta': {'field': key}})
-        elif isinstance(message, (list, tuple)):
-            for error in message:
-                errors.append({'detail': error})
+                    if isinstance(error_description, str):
+                        error_description = [error_description]
+                    errors.extend([{'source': {'pointer': '/data/attributes/' + error_key}, 'detail': reason}
+                                   for reason in error_description])
         else:
-            errors.append({'detail': message})
+            if isinstance(message, str):
```

Comment:
In case the message contains unicode, for python 2, we generally prefer to do `isinstance(message, basestring)` instead. That's a superclass that covers both regular strings and unicode.